### PR TITLE
Add a ScrapingService to manage perform comic scraping

### PR DESCRIPTION
**Pull request by mcpierce**
## Status
**READY**

## Migrations
NO

## Description
Scraping is now initiated by a service that delegates to the ScrapingAdaptor. Caching is handled outside of the scraping adaptor so it can be focused solely on interacting with the comic data source.

### CODE REVIEW

File: comixed-scrapers/src/test/java/org/comixedproject/scrapers/comicvine/adaptors/ComicVineScrapingAdaptorTest.java [hunk 1]
```diff
@@ -52,50 +48,31 @@
...
   @InjectMocks private ComicVineScrapingAdaptor scrapingAdaptor;
   @Mock private ObjectFactory<ComicVineGetVolumesAction> getVolumesActionObjectFactory;
   @Mock private ComicVineGetVolumesAction getVolumesAction;
-  @Mock private ObjectFactory<ComicVineGetIssuesAction> getIssuesActionObjectFactory;
-  @Mock private ComicVineGetIssuesAction getIssuesAction;
-  @Mock private ObjectFactory<ComicVineScrapeComicAction> scrapeComicActionObjectFactory;
-  @Mock private ComicVineScrapeComicAction scrapeComicAction;
+  @Mock private ObjectFactory<ComicVineGetIssueAction> getIssuesActionObjectFactory;
+  @Mock private ComicVineGetIssueAction getIssuesAction;
+  @Mock private ObjectFactory<ComicVineGetIssueDetailsAction> getIssueDetailsActionObjectFactory;
```
Comment (BRUCELLA2) [inline comixed-scrapers/src/test/java/org/comixedproject/scrapers/comicvine/adaptors/ComicVineScrapingAdaptorTest.java]: To be coherent with the class name, i think the term issue should be in the singular :
getIssueActionObjectFactory
getIssueAction

File: comixed-services/src/main/java/org/comixedproject/service/scraping/ScrapingService.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+      if (cachedEntries != null && !cachedEntries.isEmpty()) {
+        log.debug(
+            "Decoding {} entr{}", cachedEntries.size(), cachedEntries.size() == 1 ? "y" : "ies");
+        for (int index = 0; index < cachedEntries.size(); index++) {
+          try {
+            result.add(this.objectMapper.readValue(cachedEntries.get(index), ScrapingVolume.class));
+          } catch (JsonProcessingException error) {
+            throw new ScrapingException("Failed to decode scraping volume", error);
+          }
+        }
```
Comment (BRUCELLA2) [inline comixed-services/src/main/java/org/comixedproject/service/scraping/ScrapingService.java]: Wouldn't it be possible to use an enhanced for-loop ?

File: comixed-services/src/main/java/org/comixedproject/service/scraping/ScrapingService.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        result.addAll(fetched);
+        log.debug("Encoding fetched entries");
+        final List<String> cacheEntries = new ArrayList<>();
+        for (int index = 0; index < fetched.size(); index++) {
+          try {
+            cacheEntries.add(this.objectMapper.writeValueAsString(fetched.get(index)));
+          } catch (JsonProcessingException error) {
+            throw new ScrapingException("Failed to encoded scraping volume", error);
+          }
+        }
```
Comment (BRUCELLA2) [inline comixed-services/src/main/java/org/comixedproject/service/scraping/ScrapingService.java]: Wouldn't it be possible to use an enhanced for-loop ?

**Review by BRUCELLA2 — CHANGES_REQUESTED** [review]

File: comixed-scrapers/src/test/java/org/comixedproject/scrapers/comicvine/adaptors/ComicVineScrapingAdaptorTest.java (hunk 1 above)
Comment (mcpierce) [reply comixed-scrapers/src/test/java/org/comixedproject/scrapers/comicvine/adaptors/ComicVineScrapingAdaptorTest.java]: Fixed.

File: comixed-services/src/main/java/org/comixedproject/service/scraping/ScrapingService.java (hunk 2 above)
Comment (mcpierce) [reply comixed-services/src/main/java/org/comixedproject/service/scraping/ScrapingService.java]: Yeah, it's an old habit to work with indexes. 😄

File: comixed-services/src/main/java/org/comixedproject/service/scraping/ScrapingService.java (hunk 3 above)
Comment (mcpierce) [reply comixed-services/src/main/java/org/comixedproject/service/scraping/ScrapingService.java]: Changed it to be that now.

**Review by BRUCELLA2 — APPROVED** [review]

**Pull request merged**